shaders/program/shadowcomp1.glsl

```diff
@@ -410,8 +415,19 @@ void main() {
 
         if (insideFrustrum) {
             float thisDFval = getDistanceField(vxPos);
-            if (thisDFval < 0.7) {
-                if (thisDFval > 0.1) {
+            if (thisDFval < 0.7 || nextUint() % 37 == 0) {
```

Comment:
I think this is probably very bad for coherency, meaning that most warps will have approx. one thread executing the calculations, so the majority of warps will be mostly stalled, and only a couple will be able to skip this part. I'll figure out a better way, such as marking interesting locations for GI that should also be calculated

Reply:
Will leave it up to you how to address this, its a bit beyond my current knowledge. I also find that these random air samples look pretty noisy, so they could have increased influence from neighbors